bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -531,6 +574,61 @@ private void initEncodingTypes(CaseDbConnection connection) throws SQLException,
 			releaseSingleUserCaseWriteLock();
 		}
 	}
+	
+	/**
+	 * Set up or update the hasChildren map using the tsk_objects table.
+	 * 
+	 * @param connection
+	 * @throws TskCoreException 
+	 */
+	private void populateHasChildrenMap(CaseDbConnection connection) throws TskCoreException {
+		long timestamp = System.currentTimeMillis();
+		
+		
+		Statement statement = null;
+		ResultSet resultSet = null;
+		acquireSingleUserCaseWriteLock();
+		try {
+			statement = connection.createStatement();
+			resultSet = statement.executeQuery("select distinct par_obj_id from tsk_objects"); //NON-NLS
+			
+			synchronized (this) {
+				if (hasChildrenBitSetMap == null) {
+					hasChildrenBitSetMap = new HashMap<Long, SparseBitSet>();
+				}
+				while(resultSet.next()) {
+					setHasChildren(resultSet.getLong("par_obj_id"));
+				}
+			}
+			long delay = System.currentTimeMillis() - timestamp;
+			logger.log(Level.INFO, "Time to initialize parent node cache: {0} ms", delay); //NON-NLS
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error populating parent node cache", ex);
+		} finally {
+			closeResultSet(resultSet);
+			closeStatement(statement);
+			releaseSingleUserCaseWriteLock();
+		}
+	}
+	
+	/**
+	 * Add the object IDs for a new data source to the has children map.
+	 * At present, we simply reload the entire table. 
+	 * TODO: Test whether it's worth doing a compound query to only add
+	 * entries from the new data source
```

Comment:
We should either remove this comment or create a story and reference it with a "TODO (JIRA-XXXX):" style comment, lest this perpetually slips through the cracks.